graylog2-server/src/main/java/org/graylog2/users/RoleServiceImpl.java

```diff
@@ -159,6 +163,24 @@ public boolean exists(String roleName) {
         }
     }
 
+    @Override
+    public Map<String, Role> findIdMap(Set<String> roleIds) throws NotFoundException {
+        final DBObject query = BasicDBObjectBuilder.start()
+                .push(ID)
+                .append("$in", roleIds.stream().map(ObjectId::new).collect(Collectors.toSet()))
+                .get();
```

Comment:
You can use the `DBQuery.in()` helper here:

```suggestion
        final DBQuery.Query query = DBQuery.in(ID, roleIds);
```

Mongojack also takes care of the `ObjectId` conversion.